browser/src/react-redux/App/SignupPage.jsx

```diff
@@ -0,0 +1,164 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import setUser from './setUser.action'
+import axios from 'axios';
+
+const initialState = {
+  confirmPassword: '',
+  email: '',
+  password: '',
+  emailError: '',
+  generalError: '',
+  passwordError: ''
+};
+
+
+class SignupPage extends Component {
+  static propTypes = {
+    history: PropTypes.object.isRequired,
+    setUser: PropTypes.func.isRequired,
+    user: PropTypes.shape({
+      _id: PropTypes.string,
+      email: PropTypes.string,
+      createdAt: PropTypes.string,
+      updatedAt: PropTypes.string,
+      __v: PropTypes.number
+    })
+  };
+
+  state = initialState
+
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  getConfirmPasswordError = () => {
+    return (this.state.confirmPassword && this.state.password !== this.state.confirmPassword)
+      ? 'Passwords do not match.'
+      : null;
+  }
+
+  isSubmitButtonEnabled = () => {
+    return !(
+      this.state.confirmPassword &&
+      this.state.password &&
+      this.state.confirmPassword &&
+      this.state.password === this.state.confirmPassword
+    )
+  }
+
+  onSubmit = (e) => {
+    this.setState(initialState);
+
+    e.preventDefault();
+
+    const user = {
+      email: this.state.email,
+      password: this.state.password
+    };
+
+    axios.post('/api/user', user)
+      .then(res => {
+        //set current user with redux store.
+        this.props.setUser(res.data.user);
+      })
+      .catch(err => {
+        if (err.response.data.error.errors) {
+          const errors = err.response.data.error.errors;
+
+          errors.email ? this.setState({ emailError: errors.email.message }) : null
+          errors.password ? this.setState({ passwordError: errors.password.message }) : null
+
+        } else {
+          this.setState({
+            generalError: err.response.data.error.message
+          });
+        }
+      });
+  }
+  componentDidUpdate() {
+    if (this.props.user) {
```

Comment:
An interesting thing is happening here. If you signup successfully, this works initially. Now that we have the user object, if we go back to the user signup page, it'll let us go back. Once we type a character in one of the fields, the component changes, and then it redirects back to the homepage. The fact that it is redirecting when a character is entered doesn't feel natural. Can you think of something that will create a better user experience here?